pkg/operator2/deployment.go

```diff
@@ -81,6 +81,24 @@ func defaultDeployment(
 	volumes, mounts = toVolumesAndMounts(syncData.idpSecrets, volumes, mounts)
 	volumes, mounts = toVolumesAndMounts(syncData.tplSecrets, volumes, mounts)
 
+	volumes = append(volumes, corev1.Volume{
```

Comment:
I should probably move the `key->path` mapping among the local `volume` struct methods to make this a bit less of a special snowflake, I'll keep the WIP status of the PR for that reason